### PR TITLE
QS: robustness fixes and unit tests

### PR DESCRIPTION
**Pull request by mr-tz**
This PR adds unit tests for Range and Slice boundary conditions and improves PE parsing robustness in quantumstrand.

### CODE REVIEW

**Comment by mr-tz** [comment]
@vee1e please review when you have a moment

File: floss/qs/main.py [hunk 1]
```diff
@@ -1811,7 +1832,10 @@ def compute_layout(slice: Slice) -> Layout:
     # If XOR key is found, apply XOR decoding
     if xor_key is not None:
         decoded_data = xor_static(slice.data, xor_key)
-        decoded_slice = Slice(buf=decoded_data, range=Range(offset=0, length=len(decoded_data)))
+        # Pad the decoded data to match the absolute offset, 
+        # so that Slice/Range logic based on absolute offsets still works.
+        decoded_buf = (b"\x00" * slice.offset) + decoded_data
+        decoded_slice = Slice(buf=decoded_buf, range=Range(offset=slice.offset, length=len(decoded_data)))
```
Comment (williballenthin) [inline floss/qs/main.py:1838]: hm good catch here. it seems inefficient to create this buffer with a bunch of NULLs as prefix, but maybe handling this another way would require a lot more code and book keeping

File: qs-viewer/src/App.tsx [hunk 2]
```diff
@@ -218,7 +221,7 @@ const App: React.FC = () => {
   };
 
   const handleFocusView = () => {
-    const noisyTags = ['#code', '#code-junk', '#duplicate', '#reloc'];
+    const noisyTags = ['#code', '#code-junk', '#duplicate', '#reloc', '#common'];
```
Comment (williballenthin) [inline qs-viewer/src/App.tsx:224]: maybe move this to a global const?

Comment (vee1e) [reply qs-viewer/src/App.tsx:224]: agreed, there's some duplication going on here so a global value would be better

**Comment by mr-tz** [comment]
sorry merged this accidentally, see #1274
